File: 0189-rotate-array/0189-rotate-array.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var rotate = function(nums, k) {
    let n=nums.length;
    k=k%n;
  reverse(nums,0,n-1);
  reverse(nums,0,k-1);
  reverse(nums,k,n-1);
    
};
var reverse=function(arr,i,j){
    while(i<j){
        let temp=arr[i];
        arr[i]=arr[j];
        arr[j]=temp;
        i++;
        j--;
    }
};

File: 0189-rotate-array/0189-rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        vector<int>ans(nums.size());
        k=k%nums.size();
        for(int i=0;i<k;i++){
            ans[i]=nums[nums.size()-k+i];
        }
        for(int i=0;i<nums.size()-k;i++){
            ans[i+k]=nums[i];
        }
        for(int i=0;i<ans.size();i++){
            nums[i]=ans[i];
        }
    }
};